lib/helpers.js

```diff
@@ -0,0 +1,76 @@
+const MAX_DELAY_BETWEEN_CALLS = 1140 * 1000; // 1140 = 19 minutes in seconds
+function sleep(delay) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, delay);
+  });
+}
+
+function getDelay(delay) {
+  const delayInt = parseInt(delay, 10);
+  if (!delayInt || delayInt < 1) {
+    // TODO: Edit Error message if config fields names will be changed
+    throw new Error('Configuration error: Delay value should be a positive integer');
+  }
+  return delayInt;
+}
+
+function getCallCount(callCount) {
+  const callCountInt = parseInt(callCount, 10);
+  if (!callCountInt || callCountInt < 1) {
+    // TODO: Edit Error message if config fields names will be changed
+    throw new Error('Configuration error: Call Count value should be a positive integer');
+  }
+  return callCountInt;
+}
+
+function getDelayBetweenCalls(delay, callCount) {
+  const delayBetweenCalls = (delay * 1000) / callCount;
+  if (delayBetweenCalls < 0) {
+    // TODO: Edit Error message if config fields names will be changed
+    throw new Error('Configuration error: Delay Between Calls should be positive value');
+  }
+  if (delayBetweenCalls > MAX_DELAY_BETWEEN_CALLS) {
+    // TODO: Edit Error message if config fields names will be changed
+    throw new Error(`Configuration error: Delay Between Calls should be less than ${MAX_DELAY_BETWEEN_CALLS} milliseconds`);
+  }
+  return delayBetweenCalls;
+}
+
+
+function getRateLimitDelay(logger, cfg) {
+  logger.info('Checking rate limit parameters...');
+  const { delay, callCount } = cfg;
+  if (callCount && !delay) {
+    // TODO: Edit Error message if config fields names will be changed
+    throw new Error('Call Count value should be used only in pair with Delay option');
+  }
+  let rateLimitDelay = null;
+  if (delay) {
+    const delayInt = getDelay(delay);
+    logger.info('Delay is set to:', delay);
+    if (callCount) {
+      const callCountInt = getCallCount(callCount);
+      logger.info('Call Count is set to:', callCountInt);
+      rateLimitDelay = getDelayBetweenCalls(delayInt, callCountInt);
+    } else {
+      rateLimitDelay = delay * 1000;
+    }
+    logger.info('rateLimitDelay is:', rateLimitDelay);
+  }
+  return rateLimitDelay;
+}
+
+async function rateLimit(logger, delay) {
+  if (delay) {
+    logger.info('Delay Between Calls is set to:', delay);
```

Comment:
I'd recommend to replace logger.info with logger.debug here, to reduce amount of logs

Reply:
I replaced all other level to debug, but left info for this line because we can see at log why exactly we are waiting for.